projects/observability/src/pages/explorer/explorer-service.test.ts

```diff
@@ -0,0 +1,74 @@
+import { fakeAsync, tick } from '@angular/core/testing';
+import { NavigationParamsType } from '@hypertrace/common';
+import {
+  FilterAttributeType,
+  FilterBuilderLookupService,
+  FilterOperator,
+  NumberFilterBuilder,
+  StringFilterBuilder
+} from '@hypertrace/components';
+import { AttributeMetadataType, MetadataService } from '@hypertrace/distributed-tracing';
+import { ExplorerService, ScopeQueryParam } from '@hypertrace/observability';
+import { createServiceFactory, mockProvider } from '@ngneat/spectator/jest';
+import { of } from 'rxjs';
+
+describe('Explorer service', () => {
+  const createService = createServiceFactory({
+    service: ExplorerService
+  });
+
+  const getAttributeMock = jest.fn((scope, attributeKey) =>
+    of(
+      attributeKey === 'duration'
+        ? {
+            name: 'duration',
+            displayName: 'Duration',
+            units: 'ms',
+            type: AttributeMetadataType.Number,
+            scope: scope,
+            requiresAggregation: false,
+            allowedAggregations: [],
+            groupable: false
+          }
+        : {
+            name: 'userIdentifier',
+            displayName: 'User ID',
+            units: '',
+            type: AttributeMetadataType.String,
+            scope: scope,
+            requiresAggregation: false,
+            allowedAggregations: [],
+            groupable: false
+          }
+    )
+  );
+
+  const lookupMock = jest.fn(filterAttributeType =>
+    filterAttributeType === FilterAttributeType.Number ? new NumberFilterBuilder() : new StringFilterBuilder()
+  );
+
+  test('creates nav params correctly', fakeAsync(() => {
+    const spectator = createService({
+      providers: [
+        mockProvider(FilterBuilderLookupService, { lookup: lookupMock }),
+        mockProvider(MetadataService, { getAttribute: getAttributeMock })
+      ]
+    });
+    spectator.service
+      .buildNavParamsWithFilters(ScopeQueryParam.EndpointTraces, [
+        { field: 'duration', operator: FilterOperator.GreaterThan, value: 200 },
+        { field: 'userIdentifier', operator: FilterOperator.Equals, value: 'test' }
+      ])
+      .subscribe(data =>
```

Comment:
If putting an expectation inside a subscription, we need to use something like a `done()` callback to validate that it was actually verified. Alternatively, you can use the rxjs testing tools with marble syntax. Example:
```typescript
    runFakeRxjs(({ expectObservable }) => {
      const spectator = buildService();
      expectObservable(spectator.service.canLoad({}, [])).toBe('(x|)', { x: true });
```

Otherwise, this test would pass if the returned observable was empty or never completed.